packages/hull-client/src/lib/firehose-kafka.js

```
@@ -0,0 +1,196 @@
+    if (IS_EXITING)
+      throw new Error(
+        "Process is shutting down. Not accepting connections anymore"
+      );
+
+    const message: FirehoseMessage = {
+      type,
+      requestId,
+      sessionId,
+      payload: body,
+      organization,
+      appId: config.id,
+      timestamp: new Date()
+    };
+    if (config.accessToken) {
+      message.accessToken = config.accessToken;
+    } else if (config.subjectType !== undefined) {
+      const claims: FirehoseMessageClaims = {
+        "io.hull.subjectType": config.subjectType
+      };
+      if (config.userClaim) claims["io.hull.asUser"] = config.userClaim;
+      if (config.accountClaim)
+        claims["io.hull.asAccount"] = config.accountClaim;
+      message.claims = claims;
+    }
+
+    const producerName = config.connectorName || "";
+
+    return getProducer(firehoseTransport, producerName).then(
+      producer => {
+        return new Promise((resolve, reject) => {
+          producer.produce(
+            kafkaTopic,
+            null,
+            Buffer.from(JSON.stringify(message)),
+            `${organization}/${message.appId}`,
+            Date.now(),
+            (err, offset) => {
+              if (err) {
+                logger.error(`incoming.${type}.error`, {
+                  err: err.message,
+                  message,
+                  topic: kafkaTopic
+                });
+                const clientError = new Error();
+                clientError.message = "Backend publication error";
+                return reject(clientError);
+              }
+              return resolve({ ok: true, offset });
+            }
+          );
+        });
+      },
+      () => {
+        const timeoutError = new Error();
+        timeoutError.message = "Backend connection timeout";
+        return Promise.reject(timeoutError);
+      }
+    );
+  };
+}
+
+function flushProducer(brokers, { producer, ready }) {
+  return ready.then(() => {
```

Not catching errors here due to dangling promises. Suggest to use try/catch+async-await pattern to avoid silent errors.